packages/airnode-protocol/contracts/rrp/requesters/RrpBeaconServer.sol

```diff
@@ -194,40 +187,46 @@ contract RrpBeaconServer is
     /// this RrpBeaconServer contract, (2) The sponsor must call
     /// `setUpdatePermissionStatus()` of this RrpBeaconServer contract to give
     /// request update permission to the caller of this method.
-    /// The template used here must specify a single point of data of type
-    /// `int256` and an additional timestamp of type `uint256` to be returned
-    /// because this is what `fulfill()` expects.
+    /// The template and additional parameters used here must specify a single
+    /// point of data of type `int256` and an additional timestamp of type
+    /// `uint256` to be returned because this is what `fulfill()` expects.
     /// This point of data must be castable to `int224` and the timestamp must
     /// be castable to `uint32`.
     /// @param templateId Template ID of the beacon to be updated
     /// @param sponsor Sponsor whose wallet will be used to fulfill this
     /// request
     /// @param sponsorWallet Sponsor wallet that will be used to fulfill this
     /// request
+    /// @param parameters Parameters provided by the requester in addition to
+    /// the parameters in the template
     function requestBeaconUpdate(
         bytes32 templateId,
         address sponsor,
-        address sponsorWallet
+        address sponsorWallet,
+        bytes calldata parameters
     ) external override {
         require(
             sponsorToUpdateRequesterToPermissionStatus[sponsor][msg.sender],
             "Caller not permitted"
         );
+        bytes32 beaconId = deriveBeaconId(templateId, parameters);
         bytes32 requestId = airnodeRrp.makeTemplateRequest(
             templateId,
             sponsor,
             sponsorWallet,
             address(this),
             this.fulfill.selector,
-            ""
+            parameters
         );
-        requestIdToTemplateId[requestId] = templateId;
+        requestIdToBeaconId[requestId] = beaconId;
         emit RequestedBeaconUpdate(
-            templateId,
+            beaconId,
             sponsor,
             msg.sender,
             requestId,
-            sponsorWallet
+            templateId,
+            sponsorWallet,
+            parameters
```

Comment:
I'm not sure if `templateId` and `parameters` are required here because they will also be logged by AirnodeRrp. I still went by my rule of emitting all arguments.